.github/workflows/snap.yaml

```diff
@@ -0,0 +1,133 @@
+name: snap
+
+on:
+  workflow_call:
+    inputs:
+      ubuntu-image:
+        required: false
+        type: string
+        default: "ubuntu-latest"
+      snapcraft-channel:
+        required: false
+        type: string
+        default: "latest/stable"
+      snap-name:
+        required: true
+        type: string
+      branch-name:
+        required: true
+        type: string
+      snapcraft-args:
+        required: false
+        type: string
+      snapcraft-build-path:
+        required: false
+        type: string
+        default: "."
+      snap-install-args:
+        required: false
+        type: string
+      enable-experimental-extensions-env:
+        required: false
+        type: boolean
+      test-script:
+        description: "Custom script to run some custom snap testing"
+        required: false
+        type: string
+      publish:
+        description: "whether to publish the snap to the store"
+        required: false
+        type: boolean
+        default: false
+      cleanup:
+        description: "whether to cleanup artifacts"
+        required: false
+        type: boolean
+        default: true
+
+
+jobs:
+  build:
+    runs-on: '${{ inputs.ubuntu-image }}'
+    outputs:
+      snap-file: ${{ steps.build-snap.outputs.snap }}
+      upload-branch-name: ${{ steps.branch-name.outputs.BRANCH_NAME }}
+    steps:
+    - name: Checkout snapcraft.yaml branch
+      uses: actions/checkout@v4
+      with:
+        ref: '${{ inputs.branch-name }}'
+    - name: Build snap
+      uses: snapcore/action-build@v1
+      id: build-snap
+      env:
+        SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS : '${{ inputs.enable-experimental-extensions-env }}'
+      with:
+        path: ${{ inputs.snapcraft-build-path }}
+        snapcraft-channel: '${{ inputs.snapcraft-channel }}'
+        snapcraft-args: '${{ inputs.snapcraft-args }}'
+    - name: Branch name
+      id: branch-name
+      run: |
+        echo "BRANCH_NAME=$(echo ${{ inputs.branch-name }} | sed "s|\/|\-|")" >> "$GITHUB_OUTPUT"
+    - name: Upload snap artifact
+      uses: actions/upload-artifact@v3
```

Comment:
Please set a 'reasonable' default retention period for the artifact. By [default it is 90days](https://github.com/actions/upload-artifact?tab=readme-ov-file#retention-period). Since the artifact really is meant to be published on the store, I'd think that 14days sounds reasonable. 
```
retention-days: 14
```